src/components/codemirror.tsx

```diff
@@ -45,22 +58,156 @@ export default function CodeMirror(props: CodeMirrorProps) {
 		});
 	};
 
-
-	useEffect(() => {
+	useSignalEffect(() => {
+		if(!isNewSaveStrat.value) return
+		if(yProviderAwarenessSignal.value === undefined) return;
+		yProviderAwarenessSignal.value.on("change", () => {
+			yProviderAwarenessSignal.value?.getStates().forEach((state) => {
+				console.log(state)
+				try{
+					if(props.persistenceState === undefined) return;
```

Comment:
add comment here - what does this imply?

Reply:
this implies that something went terribly wrong. it should throw an error instead.